docs/source/index.rst

```diff
@@ -87,6 +87,7 @@ target environment.**
 
    developers/contributing
    developers/cpp/index
+   developers/guide/index
```

Comment:
```suggestion
   developers/guide/index
   developers/cpp/index
```

? Seems more logical to first have the general guide and then the C++ specific section